server/src/main/java/io/unitycatalog/server/persist/utils/ServerPropertiesUtils.java

```diff
@@ -54,24 +41,83 @@ public String getProperty(String key) {
     return properties.getProperty(key);
   }
 
-  public S3BucketConfig getS3BucketConfig(String s3Path) {
-    URI uri = URI.create(s3Path);
-    String bucketPath = uri.getScheme() + "://" + uri.getHost();
-    return (S3BucketConfig) properties.get(bucketPath);
+  public Map<String, S3StorageConfig> getS3Configurations() {
+    Map<String, S3StorageConfig> s3BucketConfigMap = new HashMap<>();
+    int i = 0;
+    while (true) {
+      String bucketPath = properties.getProperty("s3.bucketPath." + i);
+      String region = properties.getProperty("s3.region." + i);
+      String awsRoleArn = properties.getProperty("s3.awsRoleArn." + i);
+      String accessKey = properties.getProperty("s3.accessKey." + i);
+      String secretKey = properties.getProperty("s3.secretKey." + i);
+      String sessionToken = properties.getProperty("s3.sessionToken." + i);
+      if ((bucketPath == null || region == null || awsRoleArn == null)
+          && (accessKey == null || secretKey == null || sessionToken == null)) {
+        break;
```

Comment:
Yes. I didn't try and fix/change how the server properties are handled in this PR since 1) this PR is large already and 2) at some point the storage configurations should move into the db and become first class entities so not sure how much we wanted to change this before then.

Reply:
Yeah, it can be handled separately since it is an existing issue (perhaps when moving configs to the DB) -- can you add an issue for it to track?